### ISSUE TITLE
Questions about training on unlabeled data

### ISSUE THREAD

**Issue by sry2** [zhang-haojie/wesam#15]
Appreciate for your work! As the paper claims, the training of SAM seems not requiring labels originated from datasets themselves, since all the 3 types of losses can be constructed by the 3 modules' outputs. The only module may require labels seems to be the prompt generator, described as 'Weak supervision as Prompt' in section 3.3 of the paper. However, for unlabeled data, automated prompting using the output of SAM itself also seems applicable, according to section 3.3. 
The problem is that I failed to find implementation for unlabeled data in codes you provide. All datasets classes in ```./datasets``` subdirectory seems to be built for specific labeled data. Also, training code in ```line 53``` of ```./adaptation.py```:
```python
images_weak, images_strong, bboxes, gt_masks = data
```
seems to ask data sample itself for bbox and mask labels, which are also called by ```get_prompts(...)``` method followed.

So I'm wondering if codes for unlabeled data training are not released yet or I just misunderstanding the requirement of the method in certain aspects, like, the training asks for labeled data actually?

**Comment by sry2** [comment]
Thanks for the reply. I understand, closing the issue.
